src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -361,6 +366,217 @@ namespace cryptonote
   }
   //------------------------------------------------------------------------------------------------------------------------
   template<class t_core>
+  int t_cryptonote_protocol_handler<t_core>::handle_notify_new_fluffy_block(int command, NOTIFY_NEW_FLUFFY_BLOCK::request& arg, cryptonote_connection_context& context)
+  {
+    LOG_PRINT_CCONTEXT_L2("NOTIFY_NEW_FLUFFY_BLOCK (hop " << arg.hop << ")");
+    if(context.m_state != cryptonote_connection_context::state_normal)
+      return 1;
+    
+    m_core.pause_mine();
+      
+    block new_block;
+    transaction miner_tx;
+    if(parse_and_validate_block_from_blob(arg.b.block, new_block))
+    {
+      std::list<blobdata> have_tx;
+      std::list<crypto::hash> need_tx_hashes;
+        
+      transaction tx;
+      crypto::hash tx_hash;
+      std::unordered_map<crypto::hash, blobdata> tx_hash_to_blob;        
+      BOOST_FOREACH(auto& tx_blob, arg.b.txs)
+      {
+        if(parse_and_validate_tx_from_blob(tx_blob, tx))
+        {
+          get_transaction_hash(tx, tx_hash);
```

Comment:
This can actually throw with rct (unlikely, but best guarding against it). Also return code checking.